internal/functions/deploy/deploy.go

```diff
@@ -81,12 +81,20 @@ func GetFunctionConfig(slugs []string, importMapPath string, noVerifyJWT *bool,
 	functionConfig := make(config.FunctionConfig, len(slugs))
 	for _, name := range slugs {
 		function := utils.Config.Functions[name]
+		functionDir := filepath.Join(utils.FunctionsDir, name)
 		// Precedence order: flag > config > fallback
 		if len(function.Entrypoint) == 0 {
-			function.Entrypoint = filepath.Join(utils.FunctionsDir, name, "index.ts")
+			function.Entrypoint = filepath.Join(functionDir, "index.ts")
 		}
+		// Check for deno.json or deno.jsonc in functionDir
+		denoJsonPath := filepath.Join(functionDir, "deno.json")
+		denoJsoncPath := filepath.Join(functionDir, "deno.jsonc")
 		if len(importMapPath) > 0 {
 			function.ImportMap = importMapPath
+		} else if _, err := fsys.Stat(denoJsonPath); err == nil {
+			function.ImportMap = denoJsonPath
+		} else if _, err := fsys.Stat(denoJsoncPath); err == nil {
+			function.ImportMap = denoJsoncPath
```

Comment:
Is this also the default path that deno also looks for?

Either way, we'd want to move this code to [config](https://github.com/supabase/cli/blob/develop/pkg/config/config.go#L666) so it works for serve and branching deployments too.

Reply:
Yes it works exactly like `package.json`:

> Deno will automatically detect a deno.json or deno.jsonc configuration file if it's in your current working directory or parent directories. The --config flag can be used to specify a different configuration file.

I'll move the code to config then. 👍

Reply:
Logic moved to config. ✅